lang/csharp/src/apache/main/File/DataFileConstants.cs

```diff
@@ -64,10 +61,13 @@ public class DataFileConstants
         /// <summary>
         /// Magic bytes at the beginning of an Avro data file.
         /// </summary>
-        public static byte[] Magic = { (byte)'O',
-                                       (byte)'b',
-                                       (byte)'j',
-                                       Version };
+        public static readonly byte[] Magic =
```

Comment:
The looks like an API change but I guess no one is supposed to change the magic bytes

Reply:
The change was formatting.

Reply:
> The change was formatting.

What do you mean ?

AFAIU `readonly` makes `Magic` a constant, like `final` in Java.
Until now a user can change the value of `Magic` in his/her application. After this PR it won't be possible anymore. And such application will fail to compile. This is what I meant by API change.
But I agree that no one is supposed to change the magic bytes.

Reply:
You are correct.  Do we want the backwards support to allow the change?  It appears the intention was that it is a constant hence the readonly static.  Like with most things, if it is available someone is using it.  The other option is I can mark this obsolete, and add the remark for a future change that it be made readonly.

Reply:
I am 99% sure no one is setting custom value for the `Magic` field. So I am OK to merge it as is! But only for 1.12.0!
If someone complains that we broke his/her build then we can revert if (s)he gives us a really convincing reason!